src/transaction/transaction_manager.cpp

```diff
@@ -57,7 +76,8 @@ timestamp_t TransactionManager::ReadOnlyCommitCriticalSection(TransactionContext
 
 timestamp_t TransactionManager::UpdatingCommitCriticalSection(TransactionContext *const txn, const callback_fn callback,
                                                               void *const callback_arg) {
-  common::SharedLatch::ScopedExclusiveLatch guard(&commit_latch_);
+  common::SpinLatch::ScopedSpinLatch guard(&commit_latch_);
+  blocking_commit_ = true;
```

Comment:
Looking at this, it's not immediately obvious what this does. I am thinking whether there is some abstraction (`XXLatch`) that captures this thing we are trying to do here. I would much prefer the spin latch and the boolean
flag be incorporated into a class so any future users don't have to mess with the internals. I don't have a good name for it though. @mbutrovich any ideas?

Reply:
So I kind of want to call it something along the lines of a "Nand Gate" since it essentially ensures threads can proceed through one of two code points, but not both.  As best as I can tell, there's not really a good synchronization term for what this does.